docs/root/faq/configuration/timeouts.rst

```diff
@@ -107,3 +107,10 @@ TCP
   <envoy_v3_api_field_extensions.filters.network.tcp_proxy.v3.TcpProxy.idle_timeout>`
   is the amount of time that the TCP proxy will allow a connection to exist with no upstream
   or downstream activity. The default idle timeout if not otherwise specified is *1 hour*.
+
+TLS / Transport Socket
+
+* The :ref:`transport_socket_connect_timeout <envoy_v3_api_field_config.listener.v3.FilterChain.transport_socket_connect_timeout>`
+  specifies the amount of time Envoy will wait for a downstream client to complete transport-level
+  negotiations. This can be used to limit the amount of time allowed to finish a TLS handshake
```

Comment:
Worth generalizing this description further?  The timeout covers transport socket establishment, which includes the handshake step when establishing TLS or ALPN sockets.

Reply:
Sorry, I hadn't looked at the rest of this doc until now.

It makes some sense to me to move this timeout closer to the top.  Alternatively, you could add references to the HTTP connection level timeouts and TCP timeouts sections: "See :ref:\`Transport Socket\` for other connection timeouts."

Reply:
I think that now it is even more hidden than before.